src/platform/packages/shared/kbn-esql-ast/src/parser/factories/fork.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import {
+  CompositeForkSubQueryContext,
+  ForkCommandContext,
+  ForkSubQueryCommandContext,
+  ForkSubQueryProcessingCommandContext,
+  SingleForkSubQueryCommandContext,
+} from '../../antlr/esql_parser';
+import { Builder } from '../../builder';
+import { ESQLCommand } from '../../types';
+import { createCommand, createParserFields } from '../factories';
+import { createLimitCommand } from './limit';
+import { createSortCommand } from './sort';
+import { createWhereCommand } from './where';
+
+export const createForkCommand = (ctx: ForkCommandContext): ESQLCommand<'fork'> => {
+  const command = createCommand<'fork'>('fork', ctx);
+
+  const subQueryContexts = ctx.forkSubQueries().forkSubQuery_list();
+
+  for (const subCtx of subQueryContexts) {
+    const subCommands = visitForkSubQueryContext(subCtx.forkSubQueryCommand());
+    const branch = Builder.expression.query(subCommands, createParserFields(subCtx));
+    command.args.push(branch);
+  }
+
+  return command;
+};
+
+function visitForkSubQueryContext(ctx: ForkSubQueryCommandContext) {
+  const commands = [];
+
+  let nextCtx: ForkSubQueryCommandContext = ctx;
+  while (nextCtx instanceof CompositeForkSubQueryContext) {
+    const command = visitForkSubQueryProcessingCommandContext(
+      nextCtx.forkSubQueryProcessingCommand()
+    );
+    if (command) {
+      commands.unshift(command);
+    }
+
+    nextCtx = nextCtx.forkSubQueryCommand();
+  }
+
+  if (nextCtx instanceof SingleForkSubQueryCommandContext) {
+    const command = visitForkSubQueryProcessingCommandContext(
+      nextCtx.forkSubQueryProcessingCommand()
+    );
+    if (command) {
+      commands.unshift(command);
+    }
+  }
+
+  return commands;
+}
+
+function visitForkSubQueryProcessingCommandContext(ctx: ForkSubQueryProcessingCommandContext) {
+  const whereCtx = ctx.whereCommand();
```

Comment:
So this means that every time a new sub command is being added in fork we need to add this here, can we add a comment in the `exitForkCommand` to catch this?

Reply:
Yes, great idea

Reply:
Done in https://github.com/elastic/kibana/pull/216743/commits/21a070162a5ac704bf24899e331b88280c4a5304

Reply:
Thanx!